test/quantization/test_quantized_op.py

```diff
@@ -1305,6 +1315,115 @@ def test_adaptive_avg_pool2d_nhwc(self, X, output_size_h, output_size_w):
                              msg=error_message.format(name + '.zero_point', scale,
                                                       X_hat.q_zero_point()))
 
+    @given(X=hu.tensor(shapes=hu.array_shapes(min_dims=4, max_dims=5,
+                                              min_side=1, max_side=10),
+                       qparams=hu.qparams(dtypes=torch.quint8)),
+           output_size_d=st.integers(1, 10),
+           output_size_h=st.integers(1, 10),
+           output_size_w=st.integers(1, 10))
+    def test_adaptive_avg_pool3d(self, X, output_size_d, output_size_h,
```

Comment:
there are many code duplications in these test, would be great if we can dedup some, I think dedup this one with ndhwc would be striaghtforward? also dedup with the 2d test as well if possible(maybe part of the code).

Reply:
I partially refactored to remove some dups, but the ndhwc is much harder -- will refactor later